onecodex/models/experimental.py

```diff
@@ -158,14 +158,36 @@ def results(self, json=True):
         else:
             return self.table()
 
-    def table(self):
-        """Return the complete results table for the functional analysis.
+    def table(self, annotation="all", taxa_stratified=True):
+        """Return a results table for the functional analysis.
+
+        Parameters
+        ----------
+        annotation : 'str', optional
+            If not 'all', data for which group of annotations to return, one of:
+            {'pathways', 'metacyc', 'eggnog', 'go', 'ko', 'ec', 'pfam', 'reaction'}
+        taxa_stratified : 'bool', optional
+            If False, return data only by annotation ID, ignoring taxonomic stratification
 
         Returns
         -------
-        table : `pd.DataFrame`
+        results_df : `pd.DataFrame`
             A Pandas DataFrame of the functional results.
         """
         import pandas as pd
 
-        return pd.DataFrame(self._results()["table"])
+        results_df = pd.DataFrame(self._results()["table"])
+        if annotation != "all":
```

Comment:
we should probably validate `annotation` is one of the supported choices

Reply:
done, using the enum